packages/webpack/src/config/base.js

```diff
@@ -43,17 +36,29 @@ export default class WebpackBaseConfig {
 
   get nuxtEnv() {
     return {
-      isDev: this.options.dev,
+      isDev: this.dev,
       isServer: this.isServer,
       isClient: !this.isServer,
       isModern: !!this.isModern
     }
   }
 
+  get mode() {
+    return this.dev ? 'development' : 'production'
+  }
+
+  get dev() {
+    return this.context.options.dev
+  }
+
+  get buildOpts() {
```

Comment:
This is a REALLY common usage between classes. So why not moving getter to the context and use `this.buildContext.buildOptions`?

Reply:
It was, but I feel it's still to long.
OK， I'll move